evm/src/cpu/kernel/asm/secp256k1/lift_x.asm

```diff
@@ -60,759 +60,13 @@
     ADDMOD
 %endmacro
 
-// Returns a square root of x if one exists, otherwise an undefined value.
-// Computed as x^(q+1)/4, with q the Secp base field order.
-// To replace with more efficient method using non-determinism later.
+// Non-deterministically provide the square root modulo N.
```

Comment:
It might sound like this witnesses a square root and verifies it, but it's the caller's responsibility to verify it right? (Like `sqrt_secp_base` could return a bad result even if `x` is square?) Let's emphasize it in the comment, or maybe add "unsafe" in the name.

Reply:
Yeah exactly, the square root is checked in `secp_lift_x` which is the only function using `sqrt_secp_base`. I added a comment and renamed it `sqrt_secp_base_unsafe` 👍